browser/js/common/directives/graphs/add-graph.directive.js

```diff
@@ -0,0 +1,77 @@
+app.directive('addGraph', function($http) {
+    return {
+        restrict: 'E',
+        templateUrl: 'js/common/directives/graphs/add-graph.directive.html',
+        scope: {
+            data: "=",
+            info: "=",
+            columns: "="
+        },
+        link: function(scope, ele, attrs) {
+
+        	function validateNumber(col) {
```

Comment:
Looks to me like you should have some kind of "validation" factory. Or perhaps it should be called "format discovery". Either way, moving this logic into another file and component will mean more modular code, fewer unnecessary merge conflicts, and better "testability".